### ISSUE TITLE
Import material from SD card bug: Retraction length is always imported as 0

### ISSUE THREAD

**Issue by reykada** [Ultimaker/UM2.1-Firmware#13]
Looks like the issue in source file:
https://github.com/Ultimaker/UM2.1-Firmware/blob/UM2.1_JarJar/Marlin/UltiLCD2_menu_material.cpp
Commit:
https://github.com/Ultimaker/UM2.1-Firmware/commit/3bf2f9798894f56f0f2276742fbea1cc5f2002f7

Function
```
static uint16_t strToUint16(char* str, uint8_t scaling_factor)
{
    double value = strtod(str, NULL);
    value *= scaling_factor;
    return max(0, min(value, UINT16_MAX));
}
```

takes uint8_t scaling_factor

But in function `static void lcd_menu_material_import()` it calls as:
`uint16_t retraction_length = strToUint16(c, EEPROM_RETRACTION_LENGTH_SCALE);`

`EEPROM_RETRACTION_LENGTH_SCALE` defined as:
`#define EEPROM_RETRACTION_LENGTH_SCALE 256`
in "UltiLCD2_menu_material.h"

That's why strToUint16() returns always 0.

**Comment by Ghostkeeper** [comment]
Why would that return 0 then? Let's take an example input of `"5"`.

```
>>> strToUint16("5", 256)
1280
>>> 
>>> double value = strtod("5", NULL);
>>> value
5.0
>>> value *= 256;
>>> value
1280.0
>>> min(1280.0, UINT16_MAX)
1280.0
>>> max(0, 1280.0)
1280.0
```

(disclaimer: math done from the top of my head)

**Comment by reykada** [comment]
> Why would that return 0 then? Let's take an example input of `"5"`.
> 
> ```
> >>> strToUint16("5", 256)

Because
`(uint8_t)(256) = 0
`
So scaling_factor = 0, and value = 0.

**Comment by Ghostkeeper** [comment]
Good point. So this should be fixed by making the parameter a `uint16_t` (or just an `unsigned int` and let the compiler choose from anything that is at least 16-bit).

**Comment by c3podaniel** [comment]
Is it possible to implement never-too-olds solution?
